src/syntax/FStar.Syntax.Print.fs

```diff
@@ -339,9 +339,19 @@ and term_to_string x =
             | None -> ""
             | Some t -> U.format1 "by %s" (term_to_string t) in
         U.format3 "(%s <ascribed: %s %s)" (term_to_string e) annot topt
-      | Tm_match(head, branches) ->
-        U.format2 "(match %s with\n\t| %s)"
+      | Tm_match(head, asc_opt, branches) ->
+        U.format3 "(match %s %swith\n\t| %s)"
           (term_to_string head)
+          (match asc_opt with
+           | None -> ""
+           | Some (asc, tacopt) ->
+             U.format2 "ret %s%s "
+               (match asc with
+                | Inl t -> term_to_string t
+                | Inr c -> comp_to_string c)
+               (match tacopt with
```

Comment:
Can we really have a tactic associated with a returns annotation? What would it do?